pyodide-build/pyodide_build/common.py

```diff
@@ -203,3 +202,46 @@ def search_pyodide_root(curdir: str | Path, *, max_depth: int = 5) -> Path:
     raise FileNotFoundError(
         "Could not find Pyodide root directory. If you are not in the Pyodide directory, set `PYODIDE_ROOT=<pyodide-root-directory>`."
     )
+
+
+def init_environment():
+    if os.environ.get("__LOADED_PYODIDE_ENV"):
+        return
+    os.environ["__LOADED_PYODIDE_ENV"] = "1"
+    # If we are building docs, we don't need to know the PYODIDE_ROOT
+    if "sphinx" in sys.modules:
+        os.environ["PYODIDE_ROOT"] = ""
+
+    if "PYODIDE_ROOT" not in os.environ:
+        os.environ["PYODIDE_ROOT"] = str(search_pyodide_root(os.getcwd()))
+
+    os.environ.update(get_make_environment_vars())
+    hostsitepackages = get_hostsitepackages()
+    pythonpath = [
+        hostsitepackages,
+    ]
+    os.environ["PYTHONPATH"] = ":".join(pythonpath)
+    os.environ["BASH_ENV"] = ""
+    get_unisolated_packages()
+
+
+@functools.cache
+def get_pyodide_root():
+    init_environment()
+    return Path(os.environ["PYODIDE_ROOT"])
+
+
+@functools.cache
+def get_unisolated_packages():
+    import json
+
+    if "UNISOLATED_PACKGES" in os.environ:
+        return json.loads(os.environ["UNISOLATED_PACKGES"])
+    PYODIDE_ROOT = get_pyodide_root()
+    unisolated_packages = []
+    for pkg in (PYODIDE_ROOT / "packages").glob("**/*.meta"):
```

Comment:
Shouldn't this be:

```suggestion
    for pkg in (PYODIDE_ROOT / "packages").glob("**/meta.yaml"):
```

It's interesting why it works...?